src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/FileValidator.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.IO;
+using AWS.Deploy.Common.IO;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Validates that a recipe or deployment bundle option with a FilePath typehint points to an actual file.
+    /// This can either be an absolute path to the file or relative to the project path for the current recommendation.
+    /// </summary>
+    public class FileValidator : IOptionSettingItemValidator
+    {
+        public static readonly string defaultValidationFailedMessage = "The specified file does not exist";
+
+        public string ValidationFailedMessage { get; set; } = defaultValidationFailedMessage;
+
+        ValidationResult IOptionSettingItemValidator.Validate(object input, Recommendation recommendation)
+        {
+            IFileManager fileManager = new FileManager();
+            var inputFilePath = input?.ToString() ?? string.Empty;
+
+            // Allow the user to clear the option intentionally
+            if (inputFilePath == string.Empty)
+            {
+                return ValidationResult.Valid();
+            }
+
+            // Otherwise if there is a value, verify that it points to an actual file
+            if (fileManager.Exists(inputFilePath, recommendation.GetProjectDirectory()))
```

Comment:
If the only reason we need the recommendation object is to get the project directory, which in turn is used to check the path relative to the project directory, then why don't we add the project directory to the file and directory managers as the base directory and set that at startup. That way through dependency injection, you wouldn't need the recommendation object here.

Reply:
At the 5/5/22 internal meeting, we had agreed upon passing in `recommendation`  to the `Validate` calls. 

I didn't explore fully, but my hunch is we set up the service collection relatively early compared to when we know the project directory, and generally prefer to avoid global state like this.